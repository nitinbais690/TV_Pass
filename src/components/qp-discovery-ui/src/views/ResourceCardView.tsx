import React, { useState } from 'react';

import {
    View,
    TouchableHighlight,
    ViewStyle,
    StyleProp,
    ImageStyle,
    TVParallaxProperties,
    StyleSheet,
    Platform,
    TextStyle,
} from 'react-native';
import LinearGradient, { LinearGradientProps } from 'react-native-linear-gradient';
import ResizableImage from './ResizableImage';
import { ResourceVm } from '../models/ViewModels';
import { dimensions, constants, padding, colors, USE_ABSOLUTE_VALUE, AspectRatio, ImageType } from 'qp-common-ui';
import FastImage from 'react-native-fast-image';

const cardPadding = padding.xs(USE_ABSOLUTE_VALUE);
const defaultImageAspectRatio = '0-16x9';
export const cardStyles = StyleSheet.create({
    container: {
        marginLeft: cardPadding,
        marginTop: cardPadding,
        marginBottom: cardPadding,
    },
    wrapperStyle: {
        width: dimensions.fullWidth / constants.catalogCardsPreview,
        aspectRatio: dimensions.cardAspectRatio,
        borderRadius: dimensions.cardRadius,
    },
    cardWrapperStyle: {
        flex: 1,
    },
    imageStyle: {
        borderRadius: dimensions.cardRadius,
        flex: 1,
        alignSelf: 'stretch',
    },
    overviewWrapperStyle: {
        height: '100%',
        width: '100%',
        position: 'absolute',
        top: 0,
        left: 0,
    },
    onFocusCardStyle: {
        borderStyle: 'solid',
        borderColor: colors.primary,
        borderWidth: 2,
        elevation: 4,
        transform: [{ scaleX: 1.12 }, { scaleY: 1.12 }],
    },
    gradientOverlayStyle: {
        backgroundColor: 'transparent',
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
    },
});

const defaultImageGradientProps = {
    locations: [0, 0.5, 1.0],
    colors: ['transparent', 'rgba(16, 16, 16, 0.4)', 'rgba(16, 16, 16, 0.8)'],
    start: { x: 0, y: 0.5 },
    end: { x: 0, y: 1 },
};

export interface ResourceCardStyle {
    /**
     * The style of the overall card container(View).
     */
    container?: StyleProp<ViewStyle>;
    /**
     * The style of the card container(View).
     */
    wrapperStyle?: StyleProp<ViewStyle>;
    /**
     * The style of the card container(View).
     */
    cardWrapperStyle?: StyleProp<ViewStyle>;
    /**
     * The style of the card container(Image).
     */
    imageStyle?: StyleProp<ImageStyle>;
    /**
     * The style for the card overview (View)
     */
    overviewWrapperStyle?: StyleProp<ViewStyle>;
    /**
     * The style of the linear gradient (View).
     */
    gradientOverlayStyle?: StyleProp<ViewStyle>;
    /*
     * The style of the card in focus (Only Android Tv and Firetv)
     */
    onFocusCardStyle?: StyleProp<ViewStyle>;
    /*
     * The style of the card footer container
     */
    footer?: StyleProp<ViewStyle>;
    /*
     * The style of the card footer title
     */
    footerTitle?: StyleProp<TextStyle>;
    /*
     * The style of the card footer subtitle
     */
    footerSubtitle?: StyleProp<TextStyle>;
}

export interface ResourceCardViewBaseProps<T> {
    /**
     * Called when the touch is released,
     * but not if cancelled (e.g. by a scroll that steals the responder lock).
     */
    onResourcePress?: (resource: T) => void;
    /**
     * Called when the resource card focused or blurred,
     */
    onFocusChange?: (resource: T, hasFocus: boolean, index?: number) => void;
    /**
     * Based on the this TV focus border will be shown in card
     */
    isShowTVFocus?: boolean;
    /**
     * Index of specific resource card,
     */
    resourceIndex?: number;
    /**
     * *(Apple TV only)* Object with properties to control Apple TV parallax effects.
     *
     * enabled: If true, parallax effects are enabled.  Defaults to true.
     * shiftDistanceX: Defaults to 2.0.
     * shiftDistanceY: Defaults to 2.0.
     * tiltAngle: Defaults to 0.05.
     * magnification: Defaults to 1.0.
     * pressMagnification: Defaults to 1.0.
     * pressDuration: Defaults to 0.3.
     * pressDelay: Defaults to 0.0.
     *
     * @platform tvOS
     */
    tvParallaxProperties?: TVParallaxProperties;
    /**
     * Props to apply as a linear gradient on top of image
     */
    linearGradientProps?: LinearGradientProps;
    /**
     * The style of the `ResourceCardView` component
     */
    cardStyle?: ResourceCardStyle;
    /**
     * The aspectRatio for the card view
     */
    cardAspectRatio?: AspectRatio;
    /**
     * The aspectRatio for the card view
     */
    cardImageType?: ImageType;
    /**
     * Boolean prop to toggle show/hide gradient
     */
    hideGradient?: boolean;
    /**
     * Unique identifier to identify the element in tests
     */
    testID?: string;
    /**
     * Skip resizing the images. Default is to resize the images.
     */
    skipResize?: boolean;
    /**
     * The color of the underlay that will show through when the touch is active.
     * Defaults to background color of the card.
     */
    underlayColor?: string;
    /**
     * Determines what the opacity of the wrapped view should be when touch is active.
     * The value should be between 0 and 1. Defaults to 0.85. Requires underlayColor to be set.
     */
    activeOpacity?: number;
    /**
     *
     */
    overlayView?: React.ComponentType<any> | React.ReactElement | null;
    /**
     *
     */
    footerView?: React.ComponentType<any> | React.ReactElement | null;
    /**
     * Indicates the orientation of the card
     */
    isPortrait?: boolean;
    /**
     * Width of a Card
     */
    cardWidth?: number;
}

export interface ResourceCardViewProps<T> extends ResourceCardViewBaseProps<T> {
    /**
     * Must be either `ResourceVm` or one of its extensions
     */
    resource: T;
}

export const cardViewDimensions = (cardStyle: ResourceCardStyle = {}): [number, number] => {
    const wrapperStyle = (cardStyle.wrapperStyle || cardStyles.wrapperStyle) as ViewStyle;
    const cardWidth = wrapperStyle.width as number;
    const cardHeight = wrapperStyle.aspectRatio
        ? cardWidth * wrapperStyle.aspectRatio
        : (wrapperStyle.height as number);
    return [cardWidth, cardHeight];
};

const ResourceCardView = <T extends ResourceVm>(props: ResourceCardViewProps<T>): JSX.Element => {
    const {
        resource,
        cardStyle: style = {},
        onResourcePress,
        onFocusChange,
        resourceIndex,
        cardAspectRatio = dimensions.cardAspectRatio,
        cardImageType = dimensions.cardImageType,
        hideGradient,
        testID = 'cardView',
        skipResize,
        overlayView,
        footerView,
        isShowTVFocus = true,
    } = props;
    const wrapperStyle = (style.wrapperStyle || cardStyles.wrapperStyle) as ViewStyle;
    const [cardWidth] = cardViewDimensions(style);
    const [cardHeight] = cardViewDimensions(style);
    const [isFocussed, setIsFocussed] = useState<boolean>(false);
    const underlayColor = props.underlayColor ? props.underlayColor : wrapperStyle.backgroundColor;
    const activeOpacity = props.activeOpacity ? props.activeOpacity : undefined;
    const aspectRatioKey = cardAspectRatio;
    const imageType = cardImageType;
    const key = resource.imageAspectRatio ? resource.imageAspectRatio : defaultImageAspectRatio;
    const imageUrl = resource.syndicationImages && resource.syndicationImages[key];
    const shouldApplyFocusStyle = Platform.OS === 'android' && Platform.isTV;
    const onCardPress = (): void => onResourcePress && onResourcePress(resource);
    const onCardFocus = (): void => {
        setIsFocussed(true);
        onFocusChange && onFocusChange(resource, true, resourceIndex);
    };
    const onCardBlur = (): void => {
        setIsFocussed(false);
        onFocusChange && onFocusChange(resource, false, resourceIndex);
    };
    return (
        <TouchableHighlight
            style={[
                cardStyles.container,
                style.container,
                isFocussed && isShowTVFocus ? style.onFocusCardStyle || cardStyles.onFocusCardStyle : undefined,
            ]}
            underlayColor={underlayColor}
            activeOpacity={activeOpacity}
            onPress={onCardPress}
            tvParallaxProperties={props.tvParallaxProperties}
            onFocus={shouldApplyFocusStyle ? onCardFocus : undefined}
            onBlur={shouldApplyFocusStyle ? onCardBlur : undefined}
            testID={testID}
            accessibilityLabel={'Card View'}>
            <View>
                <View style={[cardStyles.wrapperStyle, style.wrapperStyle]}>
                    {resource.image ? (
                        <FastImage
                            key={resource.id || ''}
                            style={[cardStyles.imageStyle, style.imageStyle]}
                            resizeMode={FastImage.resizeMode.contain}
                            source={{ uri: resource.image }}
                        />
                    ) : (
                        <ResizableImage
                            keyId={resource.type === 'channel' ? resource.ex_id || '' : resource.id || ''}
                            width={cardWidth}
                            height={cardHeight}
                            skipResize={skipResize}
                            style={[cardStyles.imageStyle, style.imageStyle]}
                            imageType={imageType}
                            aspectRatioKey={aspectRatioKey}
                            source={{ uri: imageUrl }}
                        />
                    )}

                    {overlayView && (
                        <View style={[cardStyles.overviewWrapperStyle, style.overviewWrapperStyle]}>{overlayView}</View>
                    )}
                </View>

                {!hideGradient && (
                    <LinearGradient
                        {...defaultImageGradientProps}
                        {...props.linearGradientProps}
                        style={[cardStyles.gradientOverlayStyle, style.gradientOverlayStyle]}
                    />
                )}

                {footerView && <View style={[style.footer]}>{footerView}</View>}
            </View>
        </TouchableHighlight>
    );
};

const propsAreEqual = (prevProps: any, nextProps: any): boolean => {
    return (
        prevProps.resource.id === nextProps.resource.id &&
        prevProps.resource.expiresIn === nextProps.resource.expiresIn &&
        prevProps.resource.completedPercent === nextProps.resource.completedPercent &&
        prevProps.isPortrait === nextProps.isPortrait
    );
};

export default React.memo(ResourceCardView, propsAreEqual);
