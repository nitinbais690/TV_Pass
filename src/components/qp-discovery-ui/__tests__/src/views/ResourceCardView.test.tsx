import React from 'react';
import ResourceCardView from '../../../src/views/ResourceCardView';
import { ResourceVm } from '../../../index';

import { render, fireEvent, toJSON, getByTestId } from '@testing-library/react-native';

const createMockResourceVm = (): ResourceVm => {
    const vm: ResourceVm = {
        id: '1234',
        name: 'Test card',
        images: [
            'http://example.com/images/test_0-1x1.png',
            'http://example.com/images/test_0-2x3.png',
            'http://example.com/images/test_0-16x9.png',
        ],
        type: 'movie',
    };
    return vm;
};

const vm = createMockResourceVm();

describe('ResourceCardView', () => {
    it('renders correctly with default styles', () => {
        const jsx = <ResourceCardView resource={vm} />;
        const { container } = render(jsx);
        expect(toJSON(container)).toMatchSnapshot();
    });

    it('renders correctly and invokes onPress callback', () => {
        const pressHandler = jest.fn();
        const { container } = render(<ResourceCardView resource={vm} onResourcePress={pressHandler} />);
        const wrapper = getByTestId(container, 'cardView');
        fireEvent.press(wrapper);
        expect(pressHandler).toHaveBeenCalled();
    });

    it('avoid re-renders correctly', () => {
        const jsx = <ResourceCardView resource={vm} />;
        const { container, rerender } = render(jsx);
        expect(toJSON(container)).toMatchSnapshot();
        rerender(jsx);
        expect(toJSON(container)).toMatchSnapshot();
    });

    it("renders correctly when custom wrapper style doesn't define aspect ratio", () => {
        const cardStyle = {
            wrapperStyle: {
                width: 100,
                height: 100,
                // aspectRatio: dimensions.cardAspectRatio,
                backgroundColor: 'red',
            },
        };
        const jsx = <ResourceCardView resource={vm} cardStyle={cardStyle} />;
        const { container } = render(jsx);
        expect(toJSON(container)).toMatchSnapshot();
    });

    describe('renders with the correct aspect ratio image', () => {
        it('renders correctly 1/1 aspect ratio', () => {
            const jsx = <ResourceCardView resource={vm} cardAspectRatio={1} />;
            const { container } = render(jsx);
            expect(toJSON(container)).toMatchSnapshot();
        });

        it('renders correctly 2/3 aspect ratio', () => {
            const jsx = <ResourceCardView resource={vm} cardAspectRatio={2 / 3} />;
            const { container } = render(jsx);
            expect(toJSON(container)).toMatchSnapshot();
        });

        it('renders correctly 16/9 aspect ratio', () => {
            const jsx = <ResourceCardView resource={vm} cardAspectRatio={16 / 9} />;
            const { container } = render(jsx);
            expect(toJSON(container)).toMatchSnapshot();
        });

        it('renders fallback image for an invalid aspect ratio', () => {
            const jsx = <ResourceCardView resource={vm} cardAspectRatio={8 / 4} />;
            const { container } = render(jsx);
            expect(toJSON(container)).toMatchSnapshot();
        });

        it('renders empty image when no valid image is passed', () => {
            vm.image = '';
            vm.images = [];
            vm.id = '';
            const jsx = <ResourceCardView resource={vm} cardAspectRatio={8 / 4} />;
            const { container } = render(jsx);
            expect(toJSON(container)).toMatchSnapshot();
        });
    });
});
