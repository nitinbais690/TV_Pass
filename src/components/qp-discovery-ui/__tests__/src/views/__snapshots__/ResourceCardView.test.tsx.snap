// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCardView avoid re-renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView avoid re-renders correctly 2`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders correctly when custom wrapper style doesn't define aspect ratio 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          Object {
            "backgroundColor": "red",
            "height": 100,
            "width": 100,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=200",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders correctly with default styles 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders with the correct aspect ratio image renders correctly 1/1 aspect ratio 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders with the correct aspect ratio image renders correctly 2/3 aspect ratio 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders with the correct aspect ratio image renders correctly 16/9 aspect ratio 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders with the correct aspect ratio image renders empty image when no valid image is passed 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ResourceCardView renders with the correct aspect ratio image renders fallback image for an invalid aspect ratio 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Card View"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Array [
        Object {
          "marginBottom": 15,
          "marginLeft": 15,
          "marginTop": 15,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
    testID="cardView"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1.7777777777777777,
            "backgroundColor": "#ffffff",
            "borderRadius": 0,
            "elevation": 2,
            "shadowColor": "#333333",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": 375,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 0,
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "undefined?width=750",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <BVLinearGradient
        colors={
          Array [
            0,
            1712328720,
            3423604752,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.5,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
